### ISSUE TITLE
Neo2 - layer 4 and layer 6 not working in latest version

### ISSUE THREAD

**Issue by woylie** [pqrs-org/KE-complex_modifications#1057]
I had these modifications installed before:

- Neo2 mod 3 and 4 keys (Apple keyboard)
- Neo2 layer 4
- Neo2 layer 6
- Toggle caps_lock by pressing left_shift + right_shift at the same time

I just tried to update to the latest versions available on ke-complex-modifications.pqrs.org, but after the upgrade, layer 4 and layer 6 are not working properly any more.

Using the left mod4 key does not work on all keys. For example, mod4+j (n on Neo2) correctly results in `4`, but mod4+h (s on Neo2) gives me `S` instead of `¿`.

Pressing the left mod3 and mod4 key together should get me into layer 6, however, it seems that this brings into layer 5 instead.

Reverting `Neo2 layer 4` and `Neo2 layer 6` to the previously installed versions fixes both problems. I'm using the internal MacBook Pro keyboard with these modifications.

In the same vein, the previous `Neo2 mod 3 and 4 keys (Windows keyboard)` modification worked fine in combination with a Keychron K6 keyboard, but the newer `Neo2 mod 3 and 4 keys (Windows keyboard). Rule applied to all non-Apple keyboards` modification does not bring me to layer 3 with the left mod 3 key.

**Comment by MuhammedZakir** [comment]
Try pinging the maintainers/contributors of that config. They maybe able to give better answers.

**Comment by woylie** [comment]
Sorry for pinging you directly, but you seem to be the last people who made updates to this config @jgosmann @David-Matz (can't see the Github username of last contributor for some reason).

**Comment by jgosmann** [comment]
I will have a look and see if I can figure out what the issue is. Might take a few days though because I'm fairly busy at the moment.

**Comment by woylie** [comment]
This commit breaks it for me: https://github.com/pqrs-org/KE-complex_modifications/commit/436d4f944a2a58b8295244a20f52a488a6c5c596. Before that, everything is fine.

**Comment by jgosmann** [comment]
Pinging @Kuchenaxt who seems to have made these changes (I wasn't aware of them).

**Comment by jgosmann** [comment]
@Kuchenaxt I took a look at the changes. First of all, thank you for the contribution. I was thinking of separating the rules by the vendor ID for a while now, but haven't gotten around to take a look so far. Seems like you beat me to it. 😄 However, I do not understand why the <kbd>a</kbd> and <kbd>z</kbd> keys were changed to <kbd>keypad_1</kbd> and <kbd>keypad_2</kbd>. Was this potentially an accident that should be reverted?

**Comment by jgosmann** [comment]
As @Kuchenaxt did not respond, I reverted some of the changes. @woylie see PR #1069 if you want to test the changes. So far this fixes the layer 4 problem for me. However, <kbd>mod3</kbd> is not properly working with an external keyboard for me and I have to investigate a bit further.

**Comment by jgosmann** [comment]
I just got the rule sets confused, actually <kbd>mod3</kbd> works for me.

**Comment by woylie** [comment]
I just tested your changes. It seems to work fine with the internal MacBook keyboard. However, on my external Keychron keyboard with "Neo2 mod 3 and 4 keys (third party Mac keyboard). Rule applied to all keyboards." enabled, `mod4 left` registers in the event viewer as `non_us_backslash` and results in a `^`. `mod3+4 left` then results in `↻`.

**Comment by jgosmann** [comment]
Can you try the "Neo2 mod 3 and 4 keys (third party Mac keyboard). Rule applied to all keyboards" rule instead? That behavior is what I expect from Mac keyboards that are used with the Windows keyboard rule. For some reason the key codes for <kbd>non_us_backslash</kbd> and <kbd>grave_accent_tilde</kbd> are swapped on Apple/Mac keyboards compared to Windows keyboards. It seems that your Keychron keyboard is doing the same.

**Comment by woylie** [comment]
No, that doesn't work either. I've been using previous version of the Windows keyboard rule with that keyboard.
